moveit_core/CMakeLists.txt

```diff
@@ -57,46 +56,49 @@ find_package(visualization_msgs REQUIRED)
 # TODO: Port python bindings
 # find_package(pybind11 REQUIRED)
 
-# Finds Boost Components
-include(ConfigExtras.cmake)
-
-add_subdirectory(collision_detection_bullet)
-add_subdirectory(collision_detection_fcl)
-add_subdirectory(collision_detection)
-add_subdirectory(collision_distance_field)
-add_subdirectory(constraint_samplers)
-add_subdirectory(controller_manager)
-add_subdirectory(distance_field)
-add_subdirectory(dynamics_solver)
-add_subdirectory(exceptions)
-add_subdirectory(kinematic_constraints)
-add_subdirectory(kinematics_base)
-add_subdirectory(kinematics_metrics)
-add_subdirectory(macros)
-add_subdirectory(online_signal_smoothing)
-add_subdirectory(planning_interface)
-add_subdirectory(planning_request_adapter)
-add_subdirectory(planning_scene)
-add_subdirectory(robot_model)
-add_subdirectory(robot_state)
-add_subdirectory(robot_trajectory)
-add_subdirectory(trajectory_processing)
-add_subdirectory(transforms)
-add_subdirectory(utils)
-add_subdirectory(version)
+include(ConfigExtras.cmake) # Finds Boost Components
+include(GenerateExportHeader)
+
+add_library(moveit_core SHARED)
+add_library(moveit::core ALIAS moveit_core)
+generate_export_header(moveit_core EXPORT_FILE_NAME include/moveit/export.h)
```

Comment:
should we check for `GenerateExportHeader` or make it `REQUIRED`?

Reply:
It is included on line 60, and I believe it is part of CMake itself so you don't have to find_package it. @ChrisThrasher is that how it works?

Reply:
It's not a package, it's a 1st party include module meaning it's always available and never have to `find_package` it.